frontends/web/src/routes/buy/info.tsx

```diff
@@ -59,17 +71,31 @@ class BuyInfo extends Component<Props, State> {
         }
     }
 
+    private checkSupportedCoins = () => {
+        Promise.all(
+            this.props.accounts.map((account) => (
+                apiGet(`account/${account.code}/exchange/moonpay/buy-supported`)
+                    .then(isSupported => (isSupported ? account : false))
+            ))
+        )
+        .then(results => results.filter(result => result))
+        // @ts-ignore
```

Comment:
what's ignored here exactly?

Reply:
So in the filter before it filters out all results that are false (the ones that returned false in buy-supported), so on the next line TypeScript still thinks there could be `false`

Similar issue "filter should narrow down output type"
https://github.com/microsoft/TypeScript/issues/20812

Couldn't manage as suggested to do "User-Defined Type Guards"
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
Interesting. So, something like this wouldn't work?

```ts
function isSupported(a: Account | bool): a is Account {
  return a !== false
}

.then(results => results.filter<Account>(isSupported))
```